src/Surfnet/ServiceProviderDashboard/Domain/Entity/Entity/AttributeProperty.php

```diff
@@ -0,0 +1,45 @@
+<?php
+/**
+ * Copyright 2022 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\ServiceProviderDashboard\Domain\Entity\Entity;
+
+use Surfnet\ServiceProviderDashboard\Application\Service\AttributeServiceInterface;
+use Surfnet\ServiceProviderDashboard\Domain\ValueObject\Attribute;
+
+class AttributeProperty implements AttributeInterface
+{
+    /**
+     * @var AttributeServiceInterface
+     */
+    private $attributeService;
```

Comment:
An entity should not know about the data provider it originated from. And for another rule of thumb. No application classes can ever be used in the domain layer. I suggest you create the entities in the AttributeService. Possibly useing the static constructor trick we used on one of your previous PR's.